pkg/daemon/update.go

```diff
@@ -125,6 +125,32 @@ func (dn *Daemon) deleteStaleData(oldConfig, newConfig *mcfgv1.MachineConfig) {
 	}
 }
 
+// enableUnit enables a systemd unit via symlink
+func (dn *Daemon) enableUnit(unit ignv2_2types.Unit) error {
+	// The link location
+	wantsPath := filepath.Join(dn.prefix, wantsPathSystemd, unit.Name)
+	// sanity check that we don't return an error when the link already exists
+	if _, err := os.Stat(wantsPath); err != nil {
```

Comment:
`Stat` follows the link target, but an absolute link target does not contain the prefix for the rootfs bind-mount. Depending on the link content, the result that you get here may (or may not) be incorrect.

Reply:
Hrm. Good point.

Reply:
In the end it might be most reliable to talk to pid 1 via DBus to do this?

Reply:
@cgwalters for this specific case, likely yes. But in general all files/dirs manipulation that may traverse a symlink will be subject to the de-sync above.

Reply:
Yeah.  See https://github.com/projectatomic/rpm-ostree/pull/1481 for a related issue in rpm-ostree that we fixed by resolving the symlinks inside a container via bwrap.  Since this container is privileged, it should work fine to spawn e.g. a bwrap container targeting the host and run code there if we need to.

It might work best if we created a pattern where just `/etc` and `/var` from the host are bind mounted into such a container, but the `/usr` comes from the container?

Reply:
See also https://lwn.net/Articles/723057/ for a not-yet-upstream kernel API that would help here.